openedx/core/lib/tests/test_xblock_utils.py

```diff
@@ -0,0 +1,155 @@
+        self.assertEqual(token, '123')
+
+        request_without_token = RequestFactory().get('/')
+        token = request_token(request_without_token)
+        # Test to see if the token is an uuid1 hex value
+        test_uuid = uuid.UUID(token, version=1)
+        self.assertEqual(token, test_uuid.hex)
+
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_wrap_xblock(self, store):  # pylint: disable=unused-argument
+        fragment = self.create_fragment(u"<h1>Test!</h1>")
+        test_course = CourseFactory.create(
+            org='TestX',
+            number='TS01',
+            run='2015'
+        )
+
+        test_wrap_output = wrap_xblock(
+            runtime_class='TestRuntime',
+            block=test_course,
+            view='baseview',
+            frag=fragment,
+            context=None,
+            usage_id_serializer=lambda usage_id: quote_slashes(unicode(usage_id)),
+            request_token=uuid.uuid1().get_hex()
+        )
+        self.assertIsInstance(test_wrap_output, Fragment)
+        self.assertIn('xblock-baseview', test_wrap_output.content)
+        self.assertIn('data-runtime-class="TestRuntime"', test_wrap_output.content)
+        self.assertIn('data-usage-id="i4x:;_;_TestX;_TS01;_course;_2015"', test_wrap_output.content)
+        self.assertIn('<h1>Test!</h1>', test_wrap_output.content)
+        self.assertEqual(test_wrap_output.resources[0].data, u'body {background-color:red;}')
+        self.assertEqual(test_wrap_output.resources[1].data, 'alert("Hi!");')
+
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_replace_jump_to_id_urls(self, store):  # pylint: disable=unused-argument
+        test_replace = replace_jump_to_id_urls(
+            course_id=COURSE_KEY,
+            jump_to_id_base_url='/base_url/',
+            block=CourseFactory.create(),
+            view='baseview',
+            frag=Fragment('<a href="/jump_to_id/id">'),
+            context=None
+        )
+        self.assertIsInstance(test_replace, Fragment)
+        self.assertEqual(test_replace.content, '<a href="/base_url/id">')
+
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_replace_course_urls(self, store):  # pylint: disable=unused-argument
+        test_replace = replace_course_urls(
+            course_id=COURSE_KEY,
+            block=CourseFactory.create(),
+            view='baseview',
+            frag=Fragment('<a href="/course/id">'),
+            context=None
+        )
+        self.assertIsInstance(test_replace, Fragment)
+        self.assertEqual(test_replace.content, '<a href="/courses/TestX/TS01/2015/id">')
+
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_replace_static_urls(self, store):  # pylint: disable=unused-argument
+        test_replace = replace_static_urls(
+            data_dir=None,
+            course_id=COURSE_KEY,
+            block=CourseFactory.create(),
```

Comment:
You'll still have to, in each test, test across modulestores, so I'm guessing you'll want to create one split course and one old mongo course in the setUpClass method, and then ddt across them in each tests.